### PR TITLE
events re-work / consistency

### PR DESCRIPTION
**Pull request by nmccready**

### CODE REVIEW

**Comment by nmccready** [comment]
@tombatossals @zacronos

File: src/directives/markers.js [hunk 1]
```diff
@@ -60,6 +61,7 @@ angular.module("leaflet-directive").directive('markers',
             if (!isDefined(leafletMarkers[newName])) {
                 var markerData = markersToRender[newName];
                 var marker = createMarker(markerData);
+                var layerName = (markerData? markerData.layer : undefined) || maybeLayerName; //original way takes pref
```
Comment (nmccready) [inline src/directives/markers.js]: layerName moved around to be passed into bindMarkerEvents

File: src/services/events/leafletEventsHelpers.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+angular.module("leaflet-directive")
```
Comment (nmccready) [inline src/services/events/leafletEventsHelpers.js]: boilerplate outsourced here

File: src/services/events/leafletLabelEvents.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+angular.module("leaflet-directive")
+.factory('leafletLabelEvents', function ($rootScope, $q, $log, leafletHelpers, leafletEventsHelpers) {
```
Comment (nmccready) [inline src/services/events/leafletLabelEvents.js]: labelEvents gets its own space to be shared among MarkersHelpers and PathHelpers

File: src/services/leafletEvents.js [hunk 4]
```diff
@@ -1,447 +1,6 @@
...
-                path.on(eventName, genDispatchPathEvent(eventName, logic, leafletScope, pathEvents, name));
-            }
-
-            if (Helpers.LabelPlugin.isLoaded() && isDefined(path.label)) {
-                genLabelEvents(leafletScope, logic, path, name);
-            }
-        }
-
-    };
+angular.module("leaflet-directive").factory('leafletEvents',
```
Comment (nmccready) [inline src/services/leafletEvents.js]: use all the new leafletEventHelpers to combine into this one global one.

**Comment by zacronos** [comment]
LGTM

**Pull request merged**